vcap_utils/vcap_utils/backends/base_tensorrt.py

```diff
@@ -0,0 +1,278 @@
+import numpy as np
+
+import pycuda.driver as cuda
+import tensorrt as trt
+import cv2
+
+from typing import Dict, List, Tuple, Optional, Any
+
+from vcap import (
+    Resize,
+    DETECTION_NODE_TYPE,
+    OPTION_TYPE,
+    BaseStreamState,
+    BaseBackend,
+    rect_to_coords,
+    DetectionNode,
+)
+
+
+class HostDeviceMem(object):
+    def __init__(self, host_mem, device_mem):
+        self.host = host_mem
+        self.device = device_mem
+
+    def __str__(self):
+        return "Host:\n" + str(self.host) + "\nDevice:\n" + str(self.device)
+
+    def __repr__(self):
+        return self.__str__()
+
+
+class AllocatedBuffer:
+    def __init__(self, inputs_, outputs_, bindings_, stream_):
+        self.inputs = inputs_
+        self.outputs = outputs_
+        self.bindings = bindings_
+        self.stream = stream_
+
+
+class BaseTensorRTBackend(BaseBackend):
+    def __init__(self, engine_bytes, width, height, device_id):
```

Comment:
```suggestion
    def __init__(self, engine_bytes: bytes, width: int, height: int, device_name: str):
```

This also changes `device_id` to `device_name` to more closely match other Base*Backend's